pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -456,19 +457,22 @@ public synchronized void purgeInactiveProducers() {
                 .toMillis(pulsar.getConfiguration().getBrokerDeduplicationProducerInactivityTimeoutMinutes());
 
         Iterator<java.util.Map.Entry<String, Long>> mapIterator = inactiveProducers.entrySet().iterator();
+        boolean hasInactive = false;
```

Comment:
At glance, Looks like this value is always false.

Reply:
I think I can do a more deep review later, thanks.

Reply:
Thanks, you are right. I missed a line. Fixed, PTAL.